### ISSUE TITLE
Project dependencies may have API risk issues

### ISSUE THREAD

**Issue by PyDR-Tool** [CalciferZh/minimal-hand#97]
Hi, In **minimal-hand**, inappropriate dependency versioning constraints can cause risks.

Below are the dependencies and version constraints that the project is using

```
pygame==1.9.4
open3d==0.9
tensorflow_gpu==1.14.0
transforms3d==0.3.1
keyboard==0.13.4
opencv_python==3.4.3.18
numpy==1.18.1
```

The version constraint **==** will introduce the risk of dependency conflicts because the scope of dependencies is too strict.
The version constraint **No Upper Bound** and **\*** will introduce the risk of the missing API Error because the latest version of the dependencies may remove some APIs.

After further analysis, in this project,
The version constraint of dependency **keyboard** can be changed to *>=0.9.3,<=0.13.5*.
The version constraint of dependency **numpy** can be changed to *>=1.8.0,<=1.23.0rc3*.

The above modification suggestions can reduce the dependency conflicts as much as possible, 
and introduce the latest version as much as possible without calling Error in the projects.

The invocation of the current project includes all the following methods.

<details><summary>The calling methods from the keyboard</summary>
 <pre>keyboard.is_pressed
</pre>
</details>
<details><summary>The calling methods from the numpy</summary>
 <pre>numpy.linalg.norm
</pre>
</details>
<details><summary>The calling methods from the all methods</summary>
 <pre>pygame.init
open3d.visualization.Visualizer.update_renderer
tensorflow.pad
pickle.load
open3d.visualization.Visualizer.update_geometry
tensorflow.layers.dense
zero_padding
detnet
open3d.geometry.TriangleMesh
wrappers.ModelPipeline
self.ik_model.process
tf_hmap_to_uv
pygame.display.set_mode.blit
open3d.visualization.Visualizer.create_window
tensorflow.nn.relu
cv2.VideoCapture
pickle.load.toarray
load_pkl
lmaps.append
numpy.maximum
tensorflow.ConfigProto
dmaps.append
tensorflow.norm
tensorflow.reshape
tensorflow.contrib.layers.xavier_initializer
self.cap.read
matplotlib.pyplot.show
tensorflow.cast
numpy.matmul
dense
open3d.geometry.TriangleMesh.compute_triangle_normals
tensorflow.layers.batch_normalization
numpy.sum
viewer.get_view_control.set_constant_z_far
capture.read
transforms3d.quaternions.quat2mat
pygame.time.Clock
viewer.get_view_control.convert_to_pinhole_camera_parameters
numpy.expand_dims
tensorflow.contrib.layers.l2_regularizer
tensorflow.concat.get_shape
str
utils.OneEuroFilter
xyz_to_delta
self.compute_alpha
self.dx_filter.process
open3d.geometry.TriangleMesh.compute_vertex_normals
matplotlib.pyplot.plot
pickle.dump
conv_bn
pygame.time.Clock.tick
tensorflow.expand_dims
features.get_shape.as_list
tensorflow.nn.max_pool2d
keyboard.is_pressed
tensorflow.name_scope
frame_large.np.flip.copy
data.items
numpy.abs
net_2d
open3d.visualization.Visualizer
len
utils.OneEuroFilter.process
dense_bn
matplotlib.pyplot.legend
tensorflow.gather_nd
tensorflow.argmax
LowPassFilter
tensorflow.train.Saver
pygame.surfarray.make_surface
tensorflow.stack
numpy.linalg.norm
MANOHandJoints.labels.index
viewer.get_view_control.convert_from_pinhole_camera_parameters
tensorflow.nn.sigmoid
matplotlib.pyplot.xlabel
plot_pck
inputs.get_shape
calculate_auc
numpy.linspace.reshape
pygame.display.update
tensorflow.train.Saver.restore
tensorflow.concat
open3d.utility.Vector3dVector
bottleneck
open3d.visualization.Visualizer.poll_events
self.det_model.process
tensorflow.initializers.truncated_normal
open3d.visualization.Visualizer.get_view_control
numpy.transpose
int
xyz.get_shape.as_list
hand_mesh.HandMesh.set_abs_quat
numpy.tile
cam_params.intrinsic.set_intrinsics
self.ref_T.append
open3d.utility.Vector3iVector
numpy.array
viewer.get_render_option.load_from_json
self.graph.as_default
open3d.visualization.Visualizer.get_render_option
tensorflow.shape
get_pose_tile
mano_to_mpii
xyz.get_shape
tensorflow.tile
ModelIK
tensorflow.layers.conv2d
numpy.stack
tensorflow.transpose
tensorflow.Session
frame.np.flip.copy
pygame.display.set_mode
MANOHandJoints.mesh_mapping.items
cv2.resize
open3d.geometry.TriangleMesh.paint_uniform_color
transforms3d.axangles.axangle2mat
hmaps.append
net_3d
range
pygame.display.set_caption
hand_mesh.HandMesh
MPIIHandJoints.labels.index
self.verts.copy
self.x_filter.process
tensorflow.Graph
ModelDet
numpy.linspace
wrappers.ModelPipeline.process
matplotlib.pyplot.grid
tensorflow.variable_scope
numpy.concatenate
tensorflow.constant
tensorflow.maximum
self.ref_pose.append
conv_bn_relu
capture.OpenCVCapture
live_application
matplotlib.pyplot.ylabel
matplotlib.pyplot.tight_layout
open3d.visualization.Visualizer.add_geometry
kinematics.mpii_to_mano
utils.imresize
tensorflow.placeholder
cam_params.extrinsic.copy
numpy.stack.append
self.sess.run
resnet50
open
numpy.flip
tensorflow.where
prepare_mano
numpy.finfo
network_fn
numpy.zeros
inputs.get_shape.as_list
</pre>
</details>

@developer
Could please help me check this issue?
May I pull a request to fix it?
Thank you very much.
